### PR TITLE
Bluetooth: controller: nRF53: Use SWI instead of EGU

### PR DESCRIPTION
**Pull request by cvinayak**
nRF5340 does have SWI peripheral, hence use it instead of
the single available EGU peripheral. Use of SWI will allow
controller's LLL, ULL_HIGH and ULL_LOW execution context to
be independently configured to different interrupt priority
levels.

When ULL_HIGH priority equals ULL_LOW priority, only SWI2
is used by controller. Otherwise, SWI3 is used for ULL_LOW.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

**Review by ioannisg — CHANGES_REQUESTED** [review]
I think this won't work on nRF5340 PDK, we need perhaps conditionals here.

File: subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/swi.h [hunk 1]
```diff
@@ -22,45 +16,46 @@ static inline void hal_swi_init(void)
...
+#endif /* CONFIG_BOARD_NRF5340PDK_NRF5340_CPUNET */
+
+#endif /* CONFIG_SOC_SERIES_NRF53X */
+
+static inline void hal_swi_init(void)
+{
+	/* No platform-specific initialization required. */
+}
 
 static inline void hal_swi_lll_pend(void)
```
Comment (ioannisg) [inline subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/swi.h:54]: There's a functional change, here, now: hal_swi_lll_pend() is defined unconditionally, while earlier, it was defined only under

#if !defined(CONFIG_BT_CTLR_LOW_LAT) && \
	(CONFIG_BT_CTLR_ULL_HIGH_PRIO == CONFIG_BT_CTLR_ULL_LOW_PRIO)

@cvinayak please, clarify why this change was necessary (especially for the PDK where we stick to the previous solution of using the EGU)

Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/swi.h:54]: `hal_swi_lll_pend()` was repeated in line 25, which is unnecessary when only the macro `HAL_SWI_RADIO_IRQ` needs conditional compilation.

Comment (ioannisg) [reply subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/swi.h:54]: ok, i see.

**Review by ioannisg — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]
